src/coreclr/jit/fgopt.cpp

```diff
@@ -1566,25 +1566,31 @@ bool Compiler::fgOptimizeBranchToEmptyUnconditional(BasicBlock* block, BasicBloc
         }
 
         // Optimize the JUMP to empty unconditional JUMP to go to the new target
-        FlowEdge* const newEdge = fgAddRefPred(bDest->GetTarget(), block, fgRemoveRefPred(bDest, block));
-
         switch (block->GetKind())
         {
             case BBJ_ALWAYS:
             case BBJ_CALLFINALLYRET:
+            {
+                fgRemoveRefPred(block->GetTargetEdge());
+                FlowEdge* const newEdge = fgAddRefPred(bDest->GetTarget(), block, block->GetTargetEdge());
```

Comment:
I kind of prefer

```suggestion
                FlowEdge* const oldEdge = block->GetTargetEdge();
                fgRemoveRefPred(oldEdge);
                FlowEdge* const newEdge = fgAddRefPred(bDest->GetTarget(), block, oldEdge);
```
In the long run all this probably turns into a call to a changeXXtarget method.

Reply:
Ditto for the two cases below

Reply:
I agree that looks better.

> In the long run all this probably turns into a call to a changeXXtarget method.

That's my plan -- I'll make sure those new methods use your proposed format.